tests/phpunit/tests/functions/wpUniquePrefixedId.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * Test cases for the `wp_unique_prefixed_id()` function.
+ *
+ * @package WordPress\UnitTests
+ *
+ * @since 6.4.0
+ *
+ * @group functions.php
+ * @covers ::wp_unique_prefixed_id
+ */
+class Tests_Functions_WpUniquePrefixedId extends WP_UnitTestCase {
+
+	/**
+	 * Tests that the expected unique prefixed IDs are created.
+	 *
+	 * @ticket 59681
+	 *
+	 * @dataProvider data_should_create_unique_prefixed_ids
+	 *
+	 * @param mixed $prefix   The prefix.
+	 * @param array $expected The next two expected IDs.
+	 */
+	public function test_should_create_unique_prefixed_ids( $prefix, $expected ) {
+		$id1 = wp_unique_prefixed_id( $prefix );
+		$id2 = wp_unique_prefixed_id( $prefix );
+
+		$this->assertNotSame( $id1, $id2, 'The IDs are not unique.' );
+		$this->assertSame( $expected, array( $id1, $id2 ), 'The IDs did not match the expected values.' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_should_create_unique_prefixed_ids() {
+		return array(
+			'prefix as empty string'       => array(
+				'prefix'   => '',
+				'expected' => array( '1', '2' ),
+			),
+			'prefix as (string) "0"'       => array(
+				'prefix'   => '0',
+				'expected' => array( '01', '02' ),
+			),
+			'prefix as string'             => array(
+				'prefix'   => 'test',
+				'expected' => array( 'test1', 'test2' ),
+			),
+			'prefix as string with spaces' => array(
+				'prefix'   => '   ',
+				'expected' => array( '   1', '   2' ),
+			),
+			'prefix as (string) "1"'       => array(
+				'prefix'   => '1',
+				'expected' => array( '11', '12' ),
+			),
```

Comment:
As `wp_unique_prefixed_id()` is a general function not specific to block names, it's possible that an ID may contain `.` (i.e. allowing `(string) "0.0"`, for example) and a number of other characters. [Reference](https://www.w3.org/TR/html4/types.html#:~:text=ID%20and%20NAME%20tokens%20must,periods%20(%22.%22).).

Coverage should also be added for these.

Reply:
Sure that can be added, though would be part of the prefix, and not the incremented integer value.

Reply:
Hey @costdev, commit 7a32e2600d86a513068a858e4289bc1d3a2ce04b adds 2 more happy path datasets:

* One for `'.'`
* Another for the prefix with a block name.

What do you think?

Reply:
Thanks Tonya!

If there are any other potential "gotcha" values that a potential future refactor could regress, we should consider those too.  The non-string dataset looks good for "gotcha" values, though I think we should complete the scalar datasets with `(float) 0.0` and `(float) 1.0`.

Reply:
@costdev 28cbcad9f9ceca4af4e6d784fa9b48d943a2b44f adds a double dataset.

Reply:
@costdev here's the added coverage we talked about in Make/Core slack DM 51514403078a9affd2ee03950a9814b356a8dc5f.